plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -127,6 +127,14 @@ export class GitLabClient {
       if (response.errors) {
         throw new Error(`GraphQL errors: ${JSON.stringify(response.errors)}`);
       }
+
+      if (!response.data.group?.groupMembers?.nodes) {
+        this.logger.warn(
+          `Couldn't get members for group ${groupPath}. The provided token might not have sufficient permissions`,
+        );
```

Comment:
Hmm, I'm a bit worried both that this will not be seen when needed, and that it might provide false positives. Do you know if there is a more reliable way to determine whether we have access? Perhaps something in response headers or other parts of the response body?

Reply:
Hi @Rugvip

By design, Gitlab will return a 404 error for permission issues. This is done as part of a security measure to prevent information leaking and possible enumeration.

The headers and the body don't have any informative information that I can use. 
The response body would look something like:
```
{
  "data": {
    "group": null
  }
}
```

But, since we got up to getting the members for a certain group, we can deduce that the group does exist but we might not have the right permissions to get the members.

I can change the log level from warn to error, WDYT ?